documentation/projects/proposals/search_relevancy_sandbox/20230530-implementation_plan_staging_elasticsearch_reindex_dags.md

```diff
@@ -0,0 +1,231 @@
+   step.
+5. If the index is aliased then the DAG checks if `delete_old_if_aliased=True`
+   and proceeds to run [`indices.delete`][es_py_delete] with the old index name.
+   Otherwise, the DAG ends at the previous step.
+
+[reindex]:
+  https://github.com/WordPress/openverse/blob/7427bbd4a8178d05a27e6fef07d70905ec7ef16b/ingestion_server/ingestion_server/indexer.py#L282
+[resolve]:
+  https://www.elastic.co/guide/en/elasticsearch/reference/7.12/indices-resolve-index-api.html
+[es_python_hook]:
+  https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/stable/_api/airflow/providers/elasticsearch/hooks/elasticsearch/index.html#airflow.providers.elasticsearch.hooks.elasticsearch.ElasticsearchPythonHook
+[es_py_exists_alias]:
+  https://elasticsearch-py.readthedocs.io/en/v8.8.0/api.html#elasticsearch.client.IndicesClient.exists_alias
+[es_py_put_alias]:
+  https://elasticsearch-py.readthedocs.io/en/v8.8.0/api.html#elasticsearch.client.IndicesClient.put_alias
+[es_py_resolves_index]:
+  https://elasticsearch-py.readthedocs.io/en/v8.8.0/api.html?#elasticsearch.client.IndicesClient.resolve_index
+[ing_point_alias]:
+  https://github.com/WordPress/openverse/blob/08bb0317e1110694ca4d51058bebbc1dafb4fc13/ingestion_server/ingestion_server/indexer.py#L340
+[es_py_delete]:
+  https://elasticsearch-py.readthedocs.io/en/v8.8.0/api.html?#elasticsearch.client.IndicesClient.delete
+
+<!--------------------------------------------------------------------------->
+
+### `create_proportional_by_provider_<media_type>_staging_index` DAG
+
+This DAG is intended to be used most likely with the index resulting from the
+previous DAG or from the data refresh process, that is, an index with the
+database fully indexed, as the `source_index` for the [ES
+Reindex][es_reindex_api] API.
+
+[es_reindex_api]:
+  https://www.elastic.co/guide/en/elasticsearch/reference/7.12/docs-reindex.html
+
+#### Parameters
+
+1. `source_index`: (Optional) The existing index on Elasticsearch to use as the
+   basis for the new index. If not provided, the index aliased to
+   `<media_type>-full` will be used.
+2. `percentage_of_prod`: A float indicating the proportion of items to take from
+   each provider from the total amount existing in production. E.g. `0.25` for a
+   quarter of the production documents.
+
+#### Outlined Steps
+
+1. Get the list of media count by sources from the production Openverse API
+   `https://api.openverse.engineering/v1/<media_type>/stats/`
+2. Calculate the `total_media` adding up all the counts by provider
+3. Build the `dest_index` name with the following format:
+
+```python
+ f"{media_type}-{percentage_of_prod}-percent-of-providers-{current_datetime}"
+```
+
+4. Make a a list of dictionaries mapping all the providers with the required
+   amount of items for the new index based on the provided `percentage_of_prod`
+   param.
+5. Using the [Dynamic Task Mapping][airflow_dtm] feature of Airflow, expand the
```

Comment:
This `airflow_dtm` doesn't appear to have a reference.

Reply:
Oh wow, it's weird that the check didn't catch that 😮 Thanks!